src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,327 @@
+import org.sosy_lab.java_smt.api.InterpolatingProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext.ProverOptions;
+import org.sosy_lab.java_smt.api.SolverException;
+
+public class CVC5InterpolatingProver extends CVC5AbstractProver<Term>
+    implements InterpolatingProverEnvironment<Term> {
+
+  private final FormulaManager mgr;
+  private final Set<ProverOptions> solverOptions;
+  private final int seed;
+
+  CVC5InterpolatingProver(
+      CVC5FormulaCreator pFormulaCreator,
+      ShutdownNotifier pShutdownNotifier,
+      @SuppressWarnings("unused") int randomSeed,
+      Set<ProverOptions> pOptions,
+      FormulaManager pMgr) {
+    super(pFormulaCreator, pShutdownNotifier, randomSeed, pOptions, pMgr);
+    mgr = pMgr;
+    solverOptions = pOptions;
+    seed = randomSeed;
+  }
+
+  /**
+   * Sets the same solver Options of the Original Solver to the separate solvertoSet, except for
+   * produce-interpolants which is set here. From CVC5AbstractProver Line 66
+   */
+  @Override
+  protected void setSolverOptions(int randomSeed, Set<ProverOptions> pOptions, Solver pSolver) {
+    super.setSolverOptions(randomSeed, pOptions, pSolver);
+    pSolver.setOption("produce-interpolants", "true");
+  }
+
+  @Override
+  public Term addConstraint(BooleanFormula pConstraint) throws InterruptedException {
+    Preconditions.checkState(!closed);
+    Term t = creator.extractInfo(pConstraint);
+
+    super.addConstraint(pConstraint);
+    return t; // t is not returned in the Abstract Class
+  }
+
+  @Override
+  public BooleanFormula getInterpolant(Collection<Term> pFormulasOfA)
+      throws SolverException, InterruptedException {
+    Preconditions.checkState(!closed);
+
+    if (pFormulasOfA.isEmpty()) { // Catch trivial case
+      return mgr.getBooleanFormulaManager().makeBoolean(true);
+    }
+
+    // formulasOfB := assertedFormulas - pFormulasOfA
+    Set<Term> formulasOfB =
+        assertedFormulas.stream()
+            .flatMap(c -> c.stream())
+            .filter(n -> !pFormulasOfA.contains(n))
+            .collect(ImmutableSet.toImmutableSet());
+
+    if (formulasOfB.isEmpty()) { // Catch trivial case
+      return mgr.getBooleanFormulaManager().makeBoolean(false);
+    }
+
+    // fit the Input to work with getCVC5Interpolation
+
+    ImmutableList<Collection<Term>> formAAsList = ImmutableList.of(pFormulasOfA);
+    ImmutableList<Collection<Term>> formBAsList = ImmutableList.of(formulasOfB);
+
+    Term itp = getCVC5Interpolation(formAAsList, formBAsList);
+
+    return creator.encapsulateBoolean(itp);
+  }
+
+  @Override
+  public List<BooleanFormula> getSeqInterpolants(
+      List<? extends Collection<Term>> partitionedFormulas)
+      throws SolverException, InterruptedException {
+    Preconditions.checkArgument(
+        !partitionedFormulas.isEmpty(), "at least one partition should be available.");
+
+    List<BooleanFormula> itps = Stream.of(new BooleanFormula[] {}).collect(Collectors.toList());
```

Comment:
hm, what about a plain and simple, easy to read, nicely understandable `new ArrayList<>();` instead of that `thing`?

Reply:
I was under the impression that ArrayLists were not allowed (Refaster just screamed at me in the beginning). I swapped it back to `ArrayList`.